contracts/src/v0.8/ccip/pools/GHO/UpgradeableTokenPool.sol

```diff
@@ -317,4 +325,27 @@ abstract contract UpgradeableTokenPool is IPool, OwnerIsCreator, IERC165 {
     if (IARM(i_armProxy).isCursed()) revert BadARMSignal();
     _;
   }
+
+  /// @notice Getter for proxy pool address.
+  /// @return proxyPool The proxy pool address for the given remoteChainSelector
+  function getProxyPool() public view returns (address proxyPool) {
+    assembly ("memory-safe") {
+      proxyPool := shr(96, shl(96, sload(PROXY_POOL_SLOT)))
+    }
+  }
+
+  /// @notice Setter for proxy pool address, only callable by the DAO.
+  /// @dev This router is currently set for the Eth/Arb lane, and this pool is not expected
+  /// to support any other lanes in the future - hence can be stored agnostic to chain selector.
+  /// @param proxyPool The address of the proxy pool.
+  function setProxyPool(address proxyPool) external onlyOwner {
+    _setPoolProxy(proxyPool);
+  }
+
+  function _setPoolProxy(address proxyPool) internal {
```

Comment:
is this internal function worth?

Reply:
[1de6efb](https://github.com/aave/ccip/pull/18/commits/1de6efbaef65b2f35d2cf222f122d48a7bc0d9d6)

Reply:
seems like good practice if it were needed to be exposed to the outer token pools, through, say, reinitializer. but rm-ed